bindings/experimental/distrdf/python/DistRDF/Operation.py

```diff
@@ -31,69 +29,61 @@ class Operation(object):
+    INSTANT_ACTION = "INSTANT_ACTION"
+    TRANSFORMATION = "TRANSFORMATION"
 
     def __init__(self, name, *args, **kwargs):
         """
-        Creates a new :obj:`Operation` for the given name
-        and arguments.
+        Creates a new `Operation` for the given name and arguments.
 
         Args:
             name (str): Name of the current operation.
 
-        args (list): Variable length argument list for the current
-            operation.
+            args (list): Variable length argument list for the current
+                operation.
 
-        kwargs (dict): Keyword arguments for the current operation.
+            kwargs (dict): Keyword arguments for the current operation.
         """
         self.name = name
         self.args = list(args)
         self.kwargs = kwargs
         self.op_type = self._classify_operation(name)
 
     def _classify_operation(self, name):
-        # Classifies the given operation as action or
-        # transformation and returns the type.
-
-        ops = Operation.Types
+        """
+        Classifies the given operation as action or transformation and
+        returns the type.
+        """
 
         operations_dict = {
-            'Define': ops.TRANSFORMATION,
-            'Filter': ops.TRANSFORMATION,
-            'Range': ops.TRANSFORMATION,
-            'Aggregate': ops.ACTION,
-            'Histo1D': ops.ACTION,
-            'Histo2D': ops.ACTION,
-            'Histo3D': ops.ACTION,
-            'Profile1D': ops.ACTION,
-            'Profile2D': ops.ACTION,
-            'Profile3D': ops.ACTION,
-            'Count': ops.ACTION,
-            'Min': ops.ACTION,
-            'Max': ops.ACTION,
-            'Mean': ops.ACTION,
-            'Sum': ops.ACTION,
-            'Fill': ops.ACTION,
-            'Reduce': ops.ACTION,
-            'Report': ops.ACTION,
-            'Take': ops.ACTION,
-            'Graph': ops.ACTION,
-            'Snapshot': ops.INSTANT_ACTION,
-            'Foreach': ops.INSTANT_ACTION,
-            'AsNumpy': ops.INSTANT_ACTION
+            "Define": Operation.TRANSFORMATION,
+            "Filter": Operation.TRANSFORMATION,
+            "Range": Operation.TRANSFORMATION,
+            "Aggregate": Operation.ACTION,
+            "Histo1D": Operation.ACTION,
+            "Histo2D": Operation.ACTION,
+            "Histo3D": Operation.ACTION,
+            "Profile1D": Operation.ACTION,
+            "Profile2D": Operation.ACTION,
+            "Profile3D": Operation.ACTION,
+            "Count": Operation.ACTION,
+            "Min": Operation.ACTION,
+            "Max": Operation.ACTION,
+            "Mean": Operation.ACTION,
+            "Sum": Operation.ACTION,
+            "Fill": Operation.ACTION,
+            "Reduce": Operation.ACTION,
+            "Report": Operation.ACTION,
+            "Take": Operation.ACTION,
+            "Graph": Operation.ACTION,
+            "Snapshot": Operation.INSTANT_ACTION,
+            "Foreach": Operation.INSTANT_ACTION,
```

Comment:
`Foreach` is not an instant action, right?

Reply:
It is actually, but also it's not supported currently in distrdf. This class is more of a information holder for any RDF operation. The actual list of supported operations is in the [Base](https://github.com/root-project/root/blob/22bef282645bc272812fa5ada598650975d3cddb/bindings/experimental/distrdf/python/DistRDF/Backends/Base.py#L41) backend class